src/Stratis.Bitcoin.IntegrationTests/EnvironmentMockUpHelpers/NetworkSimulator.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Wallet;
+using Xunit;
+
+namespace Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers
+{
+    public class NetworkSimulator : IDisposable
+    {
+        public List<CoreNode> Nodes { get; private set; }
+
+        private NodeBuilder nodeBuilder;
+
+        public NetworkSimulator()
+        {
+            this.nodeBuilder = NodeBuilder.Create();
+        }
+
+        public void Initialize(int nodesCount)
+        {
+            this.Nodes = new List<CoreNode>();
+
+            //create nodes
+            for (int i = 0; i < nodesCount; ++i)
+            {
+                CoreNode node = this.nodeBuilder.CreateStratisPowNode(true);
```

Comment:
Will each nodes be in different folders?

Reply:
Yes, correct. 

CreateStratisPowNode has this lines: 
```
string child = this.CreateNewEmptyFolder();
var node = new CoreNode(child, new StratisBitcoinPowRunner(callback), this);
```

And CreateNewEmptyFolder generates a new test folders like 1,2,3,4...